pint/facets/plain/quantity.py

```diff
@@ -747,18 +745,56 @@ def _add_sub(self, other, op):
                 raise DimensionalityError(self._units, "dimensionless")
             return self.__class__(magnitude, units)
 
+        # Special case for logarithmic units: dB can be added to dBm, dBW, etc.
+        # Get non-multiplicative units before checking dimensionality
+        self_non_mul_units = self._get_non_multiplicative_units()
+        other_non_mul_units = other._get_non_multiplicative_units()
+
+        # Check if we're dealing with logarithmic units that can be added (dB + dBm, etc.)
+        if (
+            len(self_non_mul_units) == 1
+            and len(other_non_mul_units) == 1
+            and hasattr(
+                self._get_unit_definition(self_non_mul_units[0]), "is_logarithmic"
+            )
+            and hasattr(
+                self._get_unit_definition(other_non_mul_units[0]), "is_logarithmic"
+            )
+            and self._get_unit_definition(self_non_mul_units[0]).is_logarithmic
+            and self._get_unit_definition(other_non_mul_units[0]).is_logarithmic
+        ):
+            # Case 1: both are the same logarithmic unit (dB + dB)
+            if self_non_mul_units[0] == other_non_mul_units[0]:
+                # Add the magnitudes directly when the same logarithmic unit
+                magnitude = op(self._magnitude, other._magnitude)
+                return self.__class__(magnitude, self._units)
+
+            # Case 2: Special handling for adding dimensionless dB to other logarithmic units
+            elif (
+                other_non_mul_units[0] == "decibel"
```

Comment:
can you check for a dimensionless log unit here instead of hard coding decibel?

this case needs a test